### DISCUSSION TITLE
Having trouble building Wix4 setup on GitHub actions

### DISCUSSION THREAD

**Discussion by poqdavid** [wixtoolset/issues#7730] [Questions]
I have tried everything and the setup builds just fine on my PC but when I try the same thing on GitHub actions it refuses to generate my .wxs file and the build fails
Any idea how I can fix this?

Setup repo:
https://github.com/poqdavid/PotatoWallSetup/
https://github.com/poqdavid/PotatoWall/actions/workflows/buildsetup.yml

Here are the logs and binlogs from the GitHub action: 
https://gist.github.com/poqdavid/f8abe40bf0ffd3249c6a0d3d57f52e0a#file-error1-log https://gist.github.com/poqdavid/f8abe40bf0ffd3249c6a0d3d57f52e0a#file-msbuild1-binlog

**Comment by BryanCrotaz** [comment]
I've seen exactly the same issue, that a build works on the dev machine but not on github actions.

I have multiple problems with HarvestDirectory https://github.com/orgs/wixtoolset/discussions/7805 (eventually I gave up and manually specified every component and file - not fun!). These included:

* $(SolutionDir) works on my dev machine but not on Github Actions
* You have to suppress registry harvesting - still don't know why (I'm harvesting files, not registry entries!) but it stopped HEAT5151 errors when harvesting

I had multiple other issues, which went away when as an experiment I tried suppressing everything. I'd love to know why this worked!

```xml
<HarvestDirectory Include="..\AgentService\bin\Release\net7.0"> <!-- don't use SolutionDir here -->
    <ComponentGroupName>AgentServiceFiles</ComponentGroupName>
    <DirectoryRefId>INSTALLFOLDER</DirectoryRefId>
    <SuppressRootDirectory>true</SuppressRootDirectory>
    <!-- dont know why but disabling harvesting registry keys makes HEAT5151 error go away here -->
    <SuppressRegistry>true</SuppressRegistry>
    <SuppressCom>true</SuppressCom>
    <HarvestDirectorySuppressFragments>true</HarvestDirectorySuppressFragments>
</HarvestDirectory>
```

**Reply by BryanCrotaz** [reply]
I don't suppose you know why some properties have the HarvestDirectory prefix on them do you?

**Reply by poqdavid** [reply]
Nope sadly I am not that good with Wix
